src/test/java/com/virtuslab/archunit/ClassStructureTestSuite.java

```diff
@@ -23,6 +25,28 @@
 
 public class ClassStructureTestSuite extends BaseArchUnitTestSuite {
 
+  @Test
+  public void abstract_classes_should_not_declare_LOG_field() {
+    classes()
+        .that()
+        .haveModifier(ABSTRACT)
+        .should(new ArchCondition<JavaClass>("not declare a LOG field, but use log() abstract method instead") {
+          @Override
+          public void check(JavaClass javaClass, ConditionEvents events) {
+            // Using getFields() and not getAllFields() to only check declared members.
+            // We don't care about potential private logger fields from superclasses.
+            javaClass.getFields().forEach(field -> {
+              if (field.getName().equals("LOG")) {
+                String message = javaClass.getFullName() + " should use `abstract Logger log()` instead of `LOG` field";
+                events.add(SimpleConditionEvent.violated(javaClass, message));
+              }
+            });
+          }
+        })
+        .because("the SLF4J logger name should reflect the name of the concrete class, not an abstract base")
```

Comment:
ok, I see the reason 😄 but you could have that included in the PR desc

Reply:
Whoops sorry 😅 here's its a matter of correctness